sparse/coo/core.py

```diff
@@ -1436,6 +1436,22 @@ def __rmatmul__(self, other):
         except NotImplementedError:
             return NotImplemented
 
+    def __array_function__(self, func, types, args, kwargs):
+        import sparse as module
+        for submodule in func.__module__.split('.')[1:]:
+            try:
+                module = getattr(module, submodule)
+            except AttributeError:
+                return NotImplemented
```

Comment:
It's worth doing `if len(func.__module__.split('.')) == 1: sparse_func = getattr(self, func.__name__)` as well.

Reply:
This is because we don't have `sparse.sum`, `sparse.mean` etc... We simply have methods.

Reply:
> This is because we don't have `sparse.sum`, `sparse.mean` etc... We simply have methods.

And why is that? I think this will in fact create some problems with `__array_function__`. Calling for example `np.transpose()` on a `sparse.COO` would then fail, and I'm certain this happens in some in Dask, for example.

Reply:
Take a look at my suggestion. ;)